language-support/java/codegen/src/main/scala/com/digitalasset/daml/lf/codegen/backend/java/inner/TemplateClass.scala

```diff
@@ -112,18 +112,28 @@ private[inner] object TemplateClass extends StrictLogging {
       templateType.build()
     }
 
+  private val ctIdClassName = ClassName get classOf[ContractId[_]]
+  private val updateClassName = ClassName get classOf[Update[_]]
+  private val createClassName = ClassName get classOf[Create[_]]
+  private def parameterizedTypeName(raw: ClassName, arg: TypeName) =
+    ParameterizedTypeName.get(raw, arg)
+
   private def generateCreateMethod(name: ClassName): MethodSpec =
     MethodSpec
       .methodBuilder("create")
       .addModifiers(Modifier.PUBLIC)
       .addAnnotation(classOf[Override])
-      .returns(classOf[javaapi.data.CreateCommand])
+      .returns(parameterizedTypeName(updateClassName, parameterizedTypeName(ctIdClassName, name)))
       .addStatement(
-        "return new $T($T.$N, this.toValue())",
+        "var command = new $T($T.$N, this.toValue())",
```

Comment:
Separating things into local variables is a lot more dangerous in generated code; maybe combine the expression back into a single one again.